core/domain/question_services_test.py

```diff
@@ -117,16 +118,40 @@ def test_get_questions_with_multi_skill_ids(self):
         self.assertEqual(
             questions[0].to_dict(), question_1.to_dict())
 
-    def test_get_questions_by_skill_ids(self):
+    def test_get_questions_by_skill_ids_with_fetch_by_mastery(self):
         question_services.create_new_question_skill_link(
             self.editor_id, self.question_id, 'skill_1', 0.3)
         question_services.create_new_question_skill_link(
             self.editor_id, self.question_id_1, 'skill_2', 0.8)
         question_services.create_new_question_skill_link(
             self.editor_id, self.question_id_2, 'skill_2', 0.5)
 
+        skill_services.create_user_skill_mastery(
+            self.new_user_id, 'skill_2', 0.5)
+
+        questions = question_services.get_questions_by_skill_ids(
```

Comment:
Test a case when we call the function with fetch_by_mastery = True but no user_id (and invalid user_id)

Reply:
Done.

Reply:
Done.